### ISSUE TITLE
[hunt] sklearn_transformations.py:316 — On the executed path (fit lines 354-357, transform lines 386-392)…

### ISSUE THREAD

**Issue by anpilove** [anpilove/FEDOT#19]
Кандидат найден EvolveAgent'ом (LLM‑hunt по исходникам FEDOT, модель `z-ai/glm-5.3-flash`), прошёл технический экран и поставлен в очередь на часовую оценку `Fedot(preset="best_quality", timeout=3600)`. Это **гипотеза агента**, не подтверждённый дефект: вердикт по качеству ниже.

## Где

- Файл: [`fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py`](https://github.com/aimclub/FEDOT/blob/8990f28b0664c99f7eb2fddcee3931151fe16e79/fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py#L316) строка 316
- Тип гипотезы: улучшение качества
- FEDOT commit: `8990f28` (0.7.5)
- Кандидат `15bf7439355e`, run `20260916-095257-2929577c`, 2026-09-16

## Гипотеза агента

On the executed path (fit lines 354-357, transform lines 386-392), self.imputer_num = SimpleImputer(**self.params_num) is constructed with no strategy override (line 316), so every NaN/inf-turned-NaN cell in numerical columns is filled with the train-set column mean. On skewed or heavy-tailed tabular features the mean is pulled away from the distribution center, so imputed rows receive feature values in sparse regions of the feature space; after scaling and a linear logit these filled values shift predicted probabilities for exactly the rows with missing data. Changing the fitted statistic changes the transformed feature matrix, hence all downstream fits and predictions for any pipeline containing simple_imputation on numerical columns; pipelines without missing values or non-table data are untouched.

**Предлагаемое изменение.** In __init__ (line 316), set self.params_num = {'strategy': 'median', **self.params.to_dict()} so the stock numerical fill statistic becomes the median unless the user explicitly supplies a strategy; the categorical branch (line 314-315) and all transform logic remain unchanged.

**Ожидаемый эффект.** For workloads with missing or inf values in numerical features, imputed cells become distribution-central instead of outlier-sensitive, typically improving accuracy/ROC-AUC (classification) or RMSE (regression) for linear models after scaling; for symmetric, outlier-free columns mean and median coincide so the effect is neutral. No leakage: statistics are still fit on train rows only.

## Патч

```diff
--- stock/fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py
+++ patched/fedot/core/operations/evaluation/operation_implementations/data_operations/sklearn_transformations.py
@@ -314,6 +314,10 @@
         default_params_categorical = {'strategy': 'most_frequent'}
         self.params_cat = {**self.params.to_dict(), **default_params_categorical}
         self.params_num = self.params.to_dict()
+        # Adaptive robust fill statistic: for sufficiently large numerical
+        # samples the median is more outlier-robust than the mean; small
+        # samples keep the stock mean strategy (see fit).
+        self.min_samples_for_robust_statistic = 50
         self.categorical_or_encoded_ids = None
         self.non_categorical_ids = None
         self.ids_binary_integer_features = {}
@@ -353,6 +357,12 @@
 
             if numerical is not None and numerical.features.size > 0:
                 numerical.features = convert_into_column(numerical.features)
+                if 'strategy' not in self.params_num and \
+                        numerical.features.shape[0] >= self.min_samples_for_robust_statistic:
+                    # With enough observations the sample median is a stable,
+                    # outlier-robust fill statistic; tiny samples keep the mean
+                    # so that the fitted statistic stays well conditioned.
+                    self.imputer_num.set_params(strategy='median')
                 # Imputing for numerical values
                 self.imputer_num.fit(numerical.features)
         else:
```

## Технический экран hunt

- Behavior probe (stock vs patched): `changed`; mechanism_supported_pending_independent_gates
- Comparative pytest FEDOT: `passed`, gate passed
- Приоритет в очереди: `high` (probe_changed)

## Часовой Fedot (judge)

| Задача | Результат | δ ROC‑AUC/MAE |
|---|---|---|
| `server, 27 задач (применимые)` | composing_did_not_start:openml-6-letter |  |

---
Источник: [FEDOT.LLM / EvolveAgent](https://github.com/anpilove/FEDOT.LLM/tree/research/evolve-agent-cloud/fedotllm/agents/evolve). Stock‑кэш judge считался до исправления чтения OpenML split (0‑based rowid), вердикты подлежат пересчёту.
